File: backend/app/services/apm_service.py
from app.schemas.apm import CriticalityScore, ReliabilityMetrics, WorkOrder, CostLedger
from datetime import datetime
import random

class ApmEngine:
    def get_criticality(self, asset_ids: list[str]) -> list[CriticalityScore]:
        return [
            CriticalityScore(
                asset_id=aid,
                business_impact="High" if random.random() > 0.5 else "Medium",
                downtime_cost_per_hour=random.uniform(100.0, 1000.0),
                criticality_rank=random.randint(1, 10)
            ) for aid in asset_ids
        ]

    def get_reliability_metrics(self, asset_ids: list[str]) -> list[ReliabilityMetrics]:
        return [
            ReliabilityMetrics(
                asset_id=aid,
                mtbf_hours=random.uniform(500, 5000),
                mttr_hours=random.uniform(2, 48),
                failure_rate_percentage=random.uniform(0.1, 5.0)
            ) for aid in asset_ids
        ]

    def get_work_orders(self, asset_id: str = None) -> list[WorkOrder]:
        orders = [
            WorkOrder(
                order_id=f"WO-{random.randint(1000,9999)}",
                asset_id=asset_id or "UNKNOWN",
                title="Preventive Maintenance",
                status="Scheduled",
                priority="High",
                created_at=datetime.utcnow(),
                scheduled_for=None,
                cost_estimate=250.0
            )
        ]
        return orders

File: src/services/apm.service.ts
import { get } from './http';

export interface WorkOrder {
  order_id: string;
  asset_id: string;
  title: string;
  status: string;
  priority: string;
  created_at: string;
  scheduled_for: string | null;
  cost_estimate: number;
}

export interface ReliabilityMetrics {
  asset_id: string;
  mtbf_hours: number;
  mttr_hours: number;
  failure_rate_percentage: number;
}

export interface CriticalityScore {
  asset_id: string;
  business_impact: string;
  downtime_cost_per_hour: number;
  criticality_rank: number;
}

export const apmService = {
  getWorkOrders: (assetId?: string) => {
    return get<WorkOrder[]>('/apm/work-orders', assetId ? { asset_id: assetId } : {});
  },
  getReliability: (assetIds: string[]) => {
    return get<ReliabilityMetrics[]>('/apm/reliability', { asset_ids: assetIds.join(',') });
  },
  getCriticality: (assetIds: string[]) => {
    return get<CriticalityScore[]>('/apm/criticality', { asset_ids: assetIds.join(',') });
  }
};
